Add coding style tools, and a link to the LLVM Coding Standards.

jpetso opened the pull request:
I think those are useful. Please review.

fffaraz commented:
"Coding Style" should not be under the "Software" category. Also there is already a coding style category after right after the books!

jpetso commented:
I know that there is a coding style category after the books (in fact, the commit contains an addition to that), however that's part of the "Resources" chapter and otherwise contains links to further reading. The section I added under "Software" contains tools to help with these tasks, I thought it would be misplaced if those went into "Resources" because they're sort of not resources but software.

Also, added another commit with Boost.Multiprecision and LibTomMath. The former I've used myself and found quite neat because of its lack of dependencies, the latter I've seen used here and there so it's probably worth a mention too.
